var/spack/repos/builtin/packages/faiss/package.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+from spack import *
+
+
+class Faiss(AutotoolsPackage, PythonPackage, CudaPackage):
+    """Faiss is a library for efficient similarity search and clustering of
+       dense vectors.
+
+      Faiss contains algorithms that search in sets of vectors of any size, up
+      to ones that possibly do not fit in RAM. It also contains supporting code
+      for evaluation and parameter tuning. Faiss is written in C++ with
+      complete wrappers for Python/numpy. Some of the most useful algorithms
+      are implemented on the GPU. It is developed by Facebook AI Research.
+    """
+
+    homepage = "https://github.com/facebookresearch/faiss"
+    url      = "https://github.com/facebookresearch/faiss/archive/v1.6.3.tar.gz"
+
+    # putting the creator's name for now
+    #  hopefully, FAISS developers will take over
+    maintainers = ['bhatiaharsh']
+
+    version('1.6.3', sha256='e1a41c159f0b896975fbb133e0240a233af5c9286c09a28fde6aefff5336e542')
+    version('1.5.3', sha256='b24d347b0285d01c2ed663ccc7596cd0ea95071f3dd5ebb573ccfc28f15f043b')
+
+    variant('python', default=False, description='Build Python bindings')
+    variant('tests',  default=False, description='Build Tests')
+
+    conflicts('+tests', when='~python', msg='+tests must be accompanied by +python')
+
+    # we don't want "extend" because we don't want to symlink to python prefix
+    depends_on('python',        when='+python', type=('build', 'run'))
+    depends_on('py-numpy',      when='+python', type=('build', 'run'))
+    depends_on('py-scipy',      when='+tests',  type=('build', 'run'))
+
+    depends_on('blas')
+    depends_on('py-setuptools', when='+python', type='build')
+    depends_on('swig',          when='+python', type='build')
+
+    # --- patch for v1.5.3 -----------------------------------------------
+    # faiss assumes that the "source directory" will always
+    # be called "faiss" (not spack-src or faiss-1.5.3)
+    # so, we will have to create a symlink to self (faiss did that in 1.6.3)
+    # and add an include path
+    patch('fixes-in-v1.5.3.patch', when='@1.5.3')
```

Comment:
Since this is a new package, do you have any use case where v1.5.3 would be needed and v1.6.3 wouldn't be a good substitute? Asking because since v1.5.3 needs to be patched, if there's no real use case for having it I would just start with adding the latest version of `faiss` and not previous ones.

Reply:
you're right. older version likely will not be used. but no harm in keeping it there, just in case..